include/swift/AST/ClangModuleLoader.h

```diff
@@ -111,6 +155,22 @@ class ClangModuleLoader : public ModuleLoader {
   /// Print the Clang type.
   virtual void printClangType(const clang::Type *type,
                               llvm::raw_ostream &os) const = 0;
+
+  /// Return the stable serialization path for the given declaration,
+  /// if there is one.
+  virtual StableSerializationPath
+  getStableSerializationPath(const clang::Decl *decl) const = 0;
+
+  /// Follow a stable serialization path.
+  virtual const clang::Decl *
+  lookupStableSerializationPath(const StableSerializationPath &path) const = 0;
+
+  /// Determine whether the given type is serializable.
+  ///
+  /// If \c checkCanonical is true, checks the canonical type,
+  /// not the given possibly-sugared type.
+  virtual bool isSerializable(const clang::Type *type,
+                              bool checkCanonical) const = 0;
```

Comment:
What do you think about removing the boolean flag and having the caller compute the canonical type if it needs to? Right now, there's only 1 caller which calls it with the second argument set to true. It feels like it is trying to do multiple things which are not really its responsibility. Instead, we could have a comment on `ClangImporter::isSerializable` giving examples for when the two serializable status (for a canonical type vs a sugared type) might be different, leaving the responsibility to the caller.

Reply:
I actually wanted to encourage the caller to think about this.